Chaincase.UI/Pages/CoinJoinPage.razor

```diff
@@ -1,51 +1,142 @@
-    @if (IsQueuedToCoinJoin)
+
+    @if (!ViewModel.IsSynced)
     {
-        <p style="color: red;">
-            Chaincase must be open to complete a CoinJoin.
-            We'll notify you when registration ends to open.
-        </p>
+        <ion-text color="primary">
+            <h2>Once your wallet has sync'd, you can pick coins to mix!</h2>
+        </ion-text>
     }
-
-    <p>₿ @RequiredBTC required to CoinJoin</p>
-
-
-    @if (IsQueuedToCoinJoin)
+    else
     {
-        <IonButton href="/send/sent" expand="block">EXIT COINJOIN </IonButton>
+        @if (ViewModel.IsQueuedToCoinJoin)
+        {
+            <p>
+                🎉 Inputs have been registered
+            </p>
+            <p class="danger-text">
+                Chaincase must be open to complete a CoinJoin.
+                We'll notify you when registration ends to open.
+            </p>
+        }
 
-    } else
-    {
-        <IonButton href="/send/sent" expand="block">COINJOIN</IonButton>
+        <ion-text color="primary">
+            <h2>
+                Amount Selected:
+                ₿ @ViewModel.AmountQueued.ToString()
+            </h2>
+        </ion-text>
 
-    }
+        @* Coin Selection*@
+        <IonButton  @onclick="ToggleSelectCoinsModal"  expand="block">
+            Coin Selection <ion-icon name="chevron-forward-outline" />
+        </IonButton>
+        <!--<a @onclick="ToggleSelectCoinsModal">Pick some coins to mix </a>-->
+        @*Coin Selection <ion-icon name="chevron-forward-outline" />*@
+        <SelectCoinsModal @bind-Visible="@IsSelectCoinsModalVisible"
+                          SelectCoinsViewModel="@ViewModel.CoinList" />
+        @* Progress Bar *@
+        <ion-text color="primary">
+            <h5>Percentage of peers queued in most advanced round </h5>
+        </ion-text>
+        <ion-progress-bar value="@ViewModel.QueuedPercentage" />
+        @* List Coin Join round info**@
+
+        <ul class="list-group">
+            <li class="badge">
+                ₿ Required to CoinJoin
+                <span class="primary">@ViewModel.RequiredBTC</span>
+            </li>
+            <li class="badge">
+                Phase
+                <span class="success"> @ViewModel.RoundPhaseState.Phase </span>
+            </li>
+            <li class="badge">
+                Registration Ends in
+                <span class="primary">@ViewModel.TimeLeftTillRoundTimeout </span>
+            </li>
+            <li class="badge">
+                Joining With at Least
+                <span class="primary">@ViewModel.PeersNeeded peers</span>
+            </li>
+            <li class="badge">
+                Registered Peers
+                <span class="primary"> @ViewModel.PeersRegistered</span>
+            </li>
+            <li class="badge">
+                Coordinator Fee
+                <span class="primary">@ViewModel.CoordinatorFeePercent % per anonymity set </span>
+            </li>
+        </ul>
```

Comment:
For theme consistency maybe use `<ion-list>` &&  `<ion-list-item>`

Reply:
Yeah good point. This was before I knew what `ion-list` was